src/core/insights/index.ts

```diff
@@ -0,0 +1,199 @@
+import * as mixpanel from "./mixpanel";
+import * as posthog from "./posthog";
+import { InsightsIdentity } from "./types";
+
+export type InsightsConfig = {
+  debug: boolean;
+  mixpanelToken: string;
+  mixpanelAutoCapture: boolean;
+  posthogApiKey: string;
+  posthogApiHost: string;
+};
+
+let debugMode = false;
+
+export const initInsights = ({
+  mixpanelToken,
+  mixpanelAutoCapture,
+  posthogApiKey,
+  posthogApiHost,
+  debug = false,
+}: InsightsConfig) => {
+  debugMode = !!debug;
+
+  try {
+    mixpanel.initMixpanel(mixpanelToken, mixpanelAutoCapture, debugMode);
+  } catch (e) {
+    if (debugMode) {
+      console.error("Failed to initialize Mixpanel", e);
+    }
+  }
+
+  try {
+    posthog.initPosthog(posthogApiKey, posthogApiHost);
+  } catch (e) {
+    if (debugMode) {
+      console.error("Failed to initialize Posthog", e);
+    }
+  }
+};
+
+export const enableDebugMode = () => {
+  debugMode = true;
+  try {
+    mixpanel.enableDebugMode();
+    posthog.enableDebugMode();
+  } catch (e) {
+    console.error("Failed to enable debug mode", e);
+  }
+};
+
+export const disableDebugMode = () => {
+  debugMode = false;
+  try {
+    mixpanel.disableDebugMode();
+    posthog.disableDebugMode();
+  } catch (e) {
+    console.error("Failed to disable debug mode", e);
+  }
+};
+
+export const identify = ({
+  userId,
+  accountId,
+  organisationId,
+  email,
+  name,
+}: InsightsIdentity) => {
+  // In very rare cases we might have a user without an account, so we'll
```

Comment:
With this case (and with the other cases annotated by this comment), should `accountId` be optional in `InsightsIdentity`?

Reply:
not really, this is an extreme edge-case where you've been _removed_ from an account and you log in afterwards. You end up in the "waiting room" and your only options are to close your account or to create a new free account, after which you have an account ID...

Reply:
Fair